### PR TITLE
Fix/focus mode modular pipeline highlight

### LINKED ISSUE #921: Focus-mode modular pipeline highlight missing
## Description

When user hovers on the Focus Mode icon in the navigation tree there’s no other highlights apart from the icon itself going brighter. The original exploration shows a connection between the tree navigation hover and the right hand side highlight that happens in the flowchart element. These are important to create the connection between the panel action and the flowchart which are interconnected and announces the action you’re about to perform. See reference links.
 
Screenshot 2022-06-10 at 14 04 1

## Acceptance criteria

- When hovering over the Focus Mode icon, not only the icon itself is highlighted, but also the respective modular pipeline element on the flow chart as seen in Zeplin.
- When hovering over the Focus Mode, not only the respective modular pipeline element on the flowchart is lit, but also the’re is a slight contrast decrease of all the rest of the pipeline. I would try .85 transparency. I think the old exploration in the reference link is too dark. Feel free to find a nice balance. The idea here is that the highlighted modular pipeline element comes to the foreground and the rest of the pipeline fades back a touch. Make it pop :)

Reference links: 

https://zpl.io/aRnYONN

image

## Checklist
- [x] Include labels so that we can categorise your issue

### PR DESCRIPTION
**Pull request by cvetankanechevska**
## Description

Resolves #921 

## Development notes

Added one more boolean property to `createInitialPipelineState` , to track when FocusModeIcon is hovered and create new action in the store for updating this value.

For the style I reused the class we already have for when node is clicked and all other nodes in the flowchart are pushed in the background with some opacity (class: `pipeline-node--faded`).

## QA notes
You can open the Gitpod link, and check the hover effect on FocusModeIcon in the left side node list.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1107"></a>

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -234,3 +234,16 @@ export function toggleFocusMode(modularPipeline) {
     modularPipeline,
   };
 }
+
+export const TOGGLE_HOVERED_FOCUS_MODE = 'TOGGLE_HOVERED_FOCUS_MODE';
+
+/**
+ * Update the value of the of the currently-active hovered node from focus mode
+ * @param {string|null} nodeHovered The node's unique identifier
```
Comment (tynandebold) [inline src/actions/index.js:242]:
```suggestion
 * @param {string|null} hoveredFocusMode The node's unique identifier
```

**Review by tynandebold — APPROVED** [review]
This looks great. Thank you!

Maybe just write a test or two covering this new addition?

**Review by rashidakanchwala — COMMENTED** [review]
thanks @cvetankanechevska -- look's great!!! :) 

nit - we can look at adding unit tests to actions, reducers, node-list and flowchart.

**Comment by cvetankanechevska** [comment]
Hey @tynandebold and @rashidakanchwala I've added tests for the action, reducer and flowchart. Not sure about the test in the node-list component, since that action is already covered with the action and reducer tests. If you have any suggestions please let me know.

**Review by Huongg — APPROVED** [review]
looks good to me, great job 😄

File: src/actions/index.js (hunk 1 above)
Comment (cvetankanechevska) [reply src/actions/index.js]: Please check my last commit for this fix. Thanks

**Review by rashidakanchwala — APPROVED** [review]
LGTM! thanks @cvetankanechevska

**Review by comym — APPROVED** [review]

File: RELEASE.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Please follow the established format:
 - Remove Tags field from dataset and parameter nodes when viewed in metadata panel. (#1100)
 - Fix keyboard support for exporting experiment runs. (#1103)
 - Fix `MemoryDataSet` not displaying on metadata panel. (#1113)
+- Add hovering on the FocusModeIcon. (#1107)
```
Comment (tynandebold) [inline RELEASE.md:23]:
```suggestion
- Enhance display of the flowchart when hovering over the FocusMode icon. (#1107)
```

**Comment by Mackay031** [comment]
Did I just mark this as resolved? If so, I apologise.

**Review by Mackay031 — APPROVED** [review]

File: RELEASE.md (hunk 2 above)
Comment (cvetankanechevska) [reply RELEASE.md]: Thanks for this, I will change it.

**Pull request merged**